### ISSUE TITLE
The required attribute is added to the <input> in 1.10, but it is not added in 1.11

### ISSUE THREAD

**Issue by listepo** [emberjs/ember.js#10754]
`{{input type='email' id='identification' required='' placeholder='Email' value=identification}}`
Ember 1.10.0
`<input id="identification" class="ember-view ember-text-field" placeholder="Email" type="email" aria-invalid="false">`
Ember 1.11.0
`<input id="identification" class="ember-view ember-text-field" placeholder="Email" required="" type="email" data-invalid="" aria-invalid="true">`

**Comment by rwjblue** [comment]
I think that the intro description is wrong, the `required` attribute is added to the `<input>` in 1.10, but it is not added in 1.11.

[JSBin: 1.10](http://emberjs.jsbin.com/rwjblue/374/edit?html,js,output)
[JSBin: 1.11](http://emberjs.jsbin.com/rwjblue/375/edit?html,js,output)

**Comment by rwjblue** [comment]
This is basically happening because `required` is a boolean attribute and `""` is a falsey value.  If you set it to something other than empty string (or false) it does work properly.

````
{{input type='email' id='identification' required=true placeholder='Email' value=identification}}
```

http://emberjs.jsbin.com/rwjblue/376/edit

**Comment by listepo** [comment]
Thank you, you have helped to solve my problem. Fixed description.

**Comment by rwjblue** [comment]
I'm going to keep this open, and give @mixonic a chance to review.  This may be a regression that we should fix (I am unsure).

**Comment by listepo** [comment]
ok

**Comment by mixonic** [comment]
After some discussion, we've decided this edge-case of boolean properties likely has very low usage. HTMLBars templates *look* like HTML, but they are in reality a different beast. Specifically, attributes are not simply set with `setAttribute` but instead use the same "prop first" logic described for bound properties ([here in the 1.11 blog post](http://emberjs.com/blog/2015/03/27/ember-1-11-0-released.html#toc_bound-attribute-syntax)). The aim here is that bound properties and literal quoted values behave consistently. A blank string via `required=''` and `required={{someBlankString}}` behaves the same in 1.11, and was inconsistent in 1.10.

If we see more developers run into this we may patch the behavior with a deprecation until 2.0, but this is the first report. Closing it for now, and we will revisit if it becomes a wider problem.

Thanks @listepo!
